ipn/ipnlocal/peerapi.go

```diff
@@ -542,7 +543,37 @@ func (h *peerAPIHandler) logf(format string, a ...any) {
 	h.ps.b.logf("peerapi: "+format, a...)
 }
 
+func (h *peerAPIHandler) validateHost(r *http.Request) error {
+	if r.Host == "peer" {
+		return nil
+	}
+	ap, err := netip.ParseAddrPort(r.Host)
+	if err != nil {
+		return err
+	}
+	hostIPPfx := netip.PrefixFrom(ap.Addr(), ap.Addr().BitLen())
+	if slices.Contains(h.ps.selfNode.Addresses, hostIPPfx) {
+		return nil
+	}
+	return fmt.Errorf("invalid Host header %q", r.Host)
```

Comment:
we know it's an ip:port now. No need for %q. The reason is:  hostIPPfx not found in self addresses.

I'd invert the if above and make the failure paths be indented, then end in "return nil"